File: kris-agent-server/src/main/java/com/kris/agent/entity/Skill.java
package com.kris.agent.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("skills")
public class Skill {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String name;

    private Integer isGlobal;

    private String skillType;

    private String sourceType;

    private String toolSchema;

    private String toolCode;

    private String promptContent;

    private String filePath;

    private Integer enabled;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: web/src/types/skill.ts
export interface SkillConfig {
  id: number
  user_id: number
  name: string
  is_global: number
  skill_type: 'tool' | 'prompt'
  source_type: string
  tool_schema?: any
  tool_code?: string
  prompt_content?: string
  file_path: string
  enabled: number
  created_at: string
}

export interface SkillFormData {
  name: string
  skillType: 'tool' | 'prompt'
  sourceType?: string
  toolSchema?: any
  toolCode?: string
  promptContent?: string
  filePath?: string
}

export interface SkillDebugForm extends SkillFormData {
  description?: string
  propertiesText?: string
  requiredText?: string
  toolCode?: string
  promptContent?: string
}
